src/lib.rs

```diff
@@ -587,6 +591,43 @@ impl<'a, 'input> Attribute<'a, 'input> {
     pub fn range(&self) -> Range<usize> {
         self.data.range.clone()
     }
+
+    /// Returns attribute's qname's range in bytes in the original document.
+    ///
+    /// ```text
+    /// <e n:attr='value'/>
+    ///    ^^^^^^
+    /// ```
+    ///
+    /// This method will return incorrect data if the attribute's qname is longer than 255 bytes,
```

Comment:
I would rather say:

To reduce memory usage the qname length is limited by `u16::MAX`.

No need to mention panics or undefined behavior.

Reply:
Done, but I also added a sentence that concretely explains what effect this limit has on the returned value because I don't think it's necessarily obvious (it probably is for `range_qname`, but not for `range_value`).  Happy to change or remove that if you'd like.